public/app.js

```diff
@@ -0,0 +1,362 @@
+     * @description Функция получения жанров
+     */
+  async getGenres() {
+      const data = await send(`https://api.spotify.com/v1/browse/categories?locale=sv_RU`);
+        if (data)
+          return data.categories.items;
+        return null
+    },
+    /**
+     * @description Функция получения плейлистов по подходящему жанру
+     * @param {string} genreId - id заданного жанра
+     */
+    async getPlaylistByGenre(genreId) {
+      const data = await send(`https://api.spotify.com/v1/browse/categories/${genreId}/playlists?limit=10`);
+        if (data)
+          return data.playlists.items;
+        return null
+    },
+    /**
+     * @description Функция получения списка песен
+     * @param {string} tracksEndPoint - ссылка для получения API ответа с информацией про песни в плейлисте
+     */
+    async getTracks(tracksEndPoint) {
+        const data = await send(`${tracksEndPoint}?limit=10`);
+        if (data)
+          return data.items;
+        return null
+    },
+    /**
+     * @description Функция получения информации о песне
+     * @param {string} tracksEndPoint - ссылка для получения API ответа с информацией про песню
+     */
+    async getTrack(trackEndPoint) {
+      const result = await send(`${trackEndPoint}`);
+      if (result)
+        return result
+      return null
+    },
+    /**
+     * @description Функция получения информации о плейлисте
+     * @param {string} id - id плейлиста
+     */
+    async getPlaylist(id) {
+      const result = await send(`https://api.spotify.com/v1/albums/${id}`);
+      if (result)
+        return result
+      return null
+    }
+  }
+})();
+
+
+const UIController = (function() {
+
+  const getElementTemplate = (type, name, img, artist) => {
+
+    return (
+      `
+    <div class="${type}Photo ">
+      <img src="${img}" class="songPhoto photo" alt="">
+    </div>
+    <div class="${type}Name">
+        <label for="boxes" class="SongTitle"><b>${name}</b></label>
+    </div>
+    <div class="${type}Artist">
+        <label for="boxes" class="PlaylistOwner">${artist}</label>
+    </div>
+    `)
+
+  }
+
+  const DOM_Elements = {
+    selectGenre: '#select_genre',
+    selectPlaylist: '#select_playlist',
+    buttonSubmit: '#btn_submit',
+    SongDetail: '#song-detail',
+    divSonglist: '.song-list'
+  }
+
+  return {
+
+    inputField : {
+    genre: document.querySelector(DOM_Elements.selectGenre),
+    playlist: document.querySelector(DOM_Elements.selectPlaylist),
+    tracks: document.querySelector(DOM_Elements.divSonglist),
+    submit: document.querySelector(DOM_Elements.buttonSubmit),
+    songDetail: document.querySelector(DOM_Elements.SongDetail)
+  }
+
+
+    ,
+    /**
+     * @description Функция создания жанров для <option>
+     * @param {string} text - название жанра
+     * @param {string}  value - id жанра
+     */
+
+    addOption(selector, text, value){
+      const options = `<option value="${value}">${text}</option>`;
+      if (selector == 'Genre')
```

Comment:
если тут всегда selector принимает только 2 значения genre и playlist. то можно написать так
```suggestion
     this.inputField[selector.toLowerCase()]?.insertAdjacentHTML('beforeend', options);
```